Graphics/Slicer/Machine/Infill.hs

```diff
@@ -80,65 +80,68 @@ infillLineSegInside contour childContours line
                   saneIntersection _ (Left NoIntersection)         _ = Nothing
                   saneIntersection _ (Right PParallel)             _ = Nothing
                   saneIntersection _ (Right PAntiParallel)         _ = Nothing
+                  -- Since every stop point of one line segment in a contour should be the same as the next start point...
+                  -- only count the first start point, when going in one direction..
                   saneIntersection _                          (Left (HitStartPoint _ point)) (Left (HitEndPoint   _ _)) = Just point
                   saneIntersection (Left (HitStartPoint _ _)) (Left (HitEndPoint   _ _    )) _                          = Nothing
+                  -- and only count the first start point, when going in the other direction.
                   saneIntersection _                          (Left (HitEndPoint   _ _    )) (Left (HitStartPoint _ _)) = Nothing
                   saneIntersection (Left (HitEndPoint   _ _)) (Left (HitStartPoint _ point)) _                          = Just point
+                  -- FIXME: we should 'stitch out' colinear segments, not just ignore them.
                   saneIntersection (Right PColinear)          (Left (HitStartPoint _ _    )) _                          = Nothing
                   saneIntersection _                          (Left (HitEndPoint   _ _    )) (Right PColinear)          = Nothing
                   saneIntersection _                          (Right PColinear)              _                          = Nothing
                   saneIntersection (Left NoIntersection)      (Left (HitEndPoint   _ point)) (Left NoIntersection)      = Just point
                   saneIntersection r1 r2 r3 = error $ "insane result of intersecting a line (" <> show myline <> ") with a contour " <> show c <> "\n" <> show r1 <> "\n" <> show r2 <> "\n" <> show r3 <> "\n"
-                  -- FIXME: we should 'stitch out' colinear segments, not just ignore them.
 
 -- Generate lines over entire print area, where each one is aligned with a -1 slope.
--- FIXME: other ways to only generate covering lines over the given contour?
 -- FIXME: assumes we're in positive space.
-coveringLineSegsNegative :: Contour -> ℝ -> [LineSeg]
-coveringLineSegsNegative (PointSequence contourPoints) ls = flip LineSeg s . f <$> [-xMin,-xMin+lsX..xMax]
-    where s = Point2 (xMaxOutside,yMaxOutside)
-          f v = Point2 (v,0)
+coveringLineSegsNegative :: Contour -> ℝ -> [PLine2]
+coveringLineSegsNegative (PointSequence contourPoints) ls = eToPLine2 . flip LineSeg slope . f <$> [0,lss..(xMax-xMinRaw)+(yMax-yMin)+lss]
```

Comment:
How is this function different than `coveringLineSegsNegative`? Can you rewrite the two of them as partially applied versions of some new `converingLineSegsHelper`? Otherwise you'll need to keep the two in sync forever, which is a great opportunity for bugs to pop up.

Reply:
Tried to cover this a bit better with comments.